apps/api/src/Controller/DefaultController.php

```diff
@@ -376,6 +408,47 @@ private function doGreencheck($url, $ip, $browser, $source = 'api', $blind = fal
         return $data['result'];
     }
 
+     /**
+     * @param $url
+     * @param $ip
+     * @param $browser
+     * @param string $source
+     * @param bool   $blind
+     *
+     * @return mixed
+     */
+    private function doGreencheckAsync($url, $ip, $browser, $source = 'api', $blind = false)
+    {
+        $promise = $this->producer->sendCommand('greencheck', JSON::encode(['key' => 0, 'url' => $url, 'ip' => $ip, 'browser' => $browser, 'source' => $source, 'blind' => $blind]), $needReply = false);
+        $cachedLookup = $this->getGreencheckResultFromCache($url);
+
+        if (null === $cachedLookup) {
+            return array(
+                "green" => false,
+                "url" => $url,
+                "data" => false
+            );
+        }
+
+        $data = JSON::decode($cachedLookup);
```

Comment:
we use json_encode in other places, so json_decode would also be logical here then naming wise?

Reply:
I wasn't sure what convention to follow here - as we use the JSON::decode form in the `DefaultController.php` file in a few places as well. I've updated to the json_decode for now, but I'm happy to change.